src/client-agent/notify.c

```diff
@@ -39,42 +39,28 @@ char *getsharedfiles()
     return (ret);
 }
 
-#ifndef WIN32
 char *get_agent_ip()
 {
     char agent_ip[IPSIZE + 1] = { '\0' };
-#if defined (__linux__) || defined (__MACH__) || defined (sun) || defined(FreeBSD) || defined(OpenBSD)
-    int sock;
-    int i;
-    static const char * REQUEST = "host_ip";
-
-    for (i = SOCK_ATTEMPTS; i > 0; --i) {
-        if (sock = control_check_connection(), sock >= 0) {
-            if (OS_SendUnix(sock, REQUEST, strlen(REQUEST)) < 0) {
-                mdebug1("Error sending msg to control socket (%d) %s", errno, strerror(errno));
-            }
-            else{
-                if (OS_RecvUnix(sock, IPSIZE, agent_ip) <= 0) {
-                    mdebug1("Error receiving msg from control socket (%d) %s", errno, strerror(errno));
-                    agent_ip[0] = '\0';
-                }
-            }
-
-            close(sock);
-            break;
-        } else {
-            mdebug2("Control module not yet available. Remaining attempts: %d", i - 1);
-            sleep(1);
+#if defined (__linux__) || defined (__MACH__) || defined (sun) || defined(FreeBSD) || defined(OpenBSD) || defined(WIN32)
+    struct sockaddr_storage sas;
+    socklen_t len = sizeof(sas);
+    int err = getsockname(agt->sock, (struct sockaddr *)&sas, &len);
+    
+    if (!err)
```

Comment:
I've searched through a lot of the notify.c changelog and couldn't find any 
`"Err" `assignment in `agent_ip` that should be here:
```c
        } else {
           mdebug1("Cannot update host IP.");
           *agent_ip = '\0';
        }
``` 
, so 'the else' 
```c
    else {
        if ((File_DateofChange(AGENTCONFIG) > 0 ) &&
                (OS_MD5_File(AGENTCONFIG, md5sum, OS_TEXT) == 0)) {
            snprintf(tmp_msg, OS_MAXSTR - OS_HEADER_SIZE, "%s%s / %s\n%s%s\n", CONTROL_HEADER,
                    getuname(), md5sum, tmp_labels, g_shared_mg_file_hash ? g_shared_mg_file_hash : no_hash_value);
        } else {
            snprintf(tmp_msg, OS_MAXSTR - OS_HEADER_SIZE, "%s%s\n%s%s\n", CONTROL_HEADER,
                    getuname(), tmp_labels, g_shared_mg_file_hash ? g_shared_mg_file_hash : no_hash_value);
        }
    }
``` 
seems to be a dead code.
That makes us think if we want to enable this or simply not consider it in search of another solution.

Reply:
> "Err" assignment 

This is an error code returned here 
https://github.com/wazuh/wazuh/blob/c168773980b67b622d44c58791eacd8c2329416b/src/wazuh_modules/wm_control.c#L230

This error code could be returned after the request through the socket. But now, it has no sense in this context.

> seems to be a dead code.

It is just generating the agent message in case the IP couldn't be found. If you look closely, the `else` block is more or less the same but without the `label_ip` field.

> That makes us think if we want to enable this or simply not consider it in search of another solution.

Sorry, not sure if I understand this.

Reply:
That's how it is! They are similar code blocks and it would be feasible to proceed, but take a look at the differences of the last change that line `169` implies (20 months ago).

![image](https://github.com/wazuh/wazuh/assets/93778492/dec7d143-3216-4bf6-ba04-94255b3eea04)
I wonder why it was not chosen at that moment to assign `"Err" `instead of `'\0'`?

Reply:
Ok looking at this again, I realize that it is not dead code!
I will then add the text in case of error.